### PR TITLE
Fix changes status and return diff information when state=touch and check_mode is set

### LINKED ISSUE #50452: File module should show return a changed=true with check_mode=true and state=touch
##### SUMMARY
Currently the file module does absolutly nothing if it is called with state=touch and check_mode=true. It should return at least a changed=true state.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

lib/ansible/modules/files/file.py

##### ANSIBLE VERSION

```
ansible 2.7.5
```

##### ADDITIONAL INFORMATION
According to this line https://github.com/ansible/ansible/blob/v2.7.5/lib/ansible/modules/files/file.py#L448 there is no code that handle the check_mode.
I had noticed some github's issues related to bad behaviour with state=touch and missing directories... but I can cite one of your developper by saying "check mode should show all changes without doing any real change"
As I can see, the touch was always triggering a changed=true state until this commit https://github.com/ansible/ansible/commit/a78cc15099d6a31e948eb5aa6f9f3cc82337f5bc that remove the {changed: true} attribute from "result" variable.
So could you please consider add a "else" case with result['changed'] = True
To show up at least that the task will perform a change.

```yaml
- name: Ensure file for specifications
      file:
        path: /tmp/test
        state: touch
```

With --check

```
TASK [Ensure file for specifications] *****************************************************************************************************************************************************************************************
ok: [host] => {"changed": false, "path": "/tmp/test", "state": "absent"}
```

Without --check

```
TASK [Ensure file for specifications] ***************************************************************************************************************************************************************************************
--- before
+++ after
@@ -1,6 +1,6 @@
 {
-    "atime": 1546440986.1151996,
-    "mtime": 1546440986.1151996,
+    "atime": 1546440986.119353,
+    "mtime": 1546440986.119352,
     "path": "/tmp/test",
-    "state": "absent"
+    "state": "touch"
 }

```

### PR DESCRIPTION
**Pull request by DSNortsev**
##### SUMMARY

Fixes #50452

* Add check_model parameter to update_timestamp_for_file function and the file timestamp
  only updates when check_mode is set to False (default option)
* Changed is set to TRUE if file does not exist or the access_time and modification_time
  option are not set to preserve

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

file

##### ADDITIONAL INFORMATION

Playbook: 
```
---
- hosts: localhost
  connection: local
  tasks:

  - name: Ensure file for specifications
    file:
      path: /tmp/test
      state: touch
```

Playbook when access_time and modification_time are preserve:
```
---
- hosts: localhost
  connection: local
  gather_facts: no
  tasks:
    - name: Ensure file for specifications
      file:
        path: /tmp/test
        state: touch
        access_time: preserve
        modification_time: preserve
```
 

Before Fix:
```paste below
ok: [localhost] => {
    "changed": false,
    "dest": "/tmp/test",
    "gid": 1001,
    "group": "dmitry",
    "invocation"
```

After Fix if file does not exist:
```
changed: [localhost] => {
    "changed": true,
    "dest": "/tmp/test",
    "invocation": {
        "module_args":
```

After Fix if file exists:
```
changed: [localhost] => {
    "changed": true,
    "dest": "/tmp/test",
    "diff": {
        "after": {
            "atime": 1557243232.714623,
            "mtime": 1557243232.714623,
            "path": "/tmp/test",
            "state": "touch"
        },
        "before": {
            "atime": 1557243223.6023092,
            "mtime": 1557243223.6023092,
            "path": "/tmp/test",
            "state": "file"
        }
    }
```
After fix if acccess_time and modification_time are set to preserve:
```
    "changed": false,
    "dest": "/tmp/test",
    "gid": 1001,
    "group": "dmitry",
    "invocation":
```

### CODE REVIEW

File: changelogs/fragments/touch_fix.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - file - Fix chnaged status and return diff information in file module when state=touch and check mode is set https://github.com/ansible/ansible/issues/50452
```
Comment (felixfontein) [inline changelogs/fragments/touch_fix.yaml]:
```suggestion
  - file - Fix changed status and return diff information in file module when ``state=touch`` and check mode is set (https://github.com/ansible/ansible/issues/50452)
```

Comment (DSNortsev) [reply changelogs/fragments/touch_fix.yaml]: Fixed. Thank you

**Comment by DSNortsev** [comment]
Open new PR: https://github.com/ansible/ansible/pull/56174

**Comment by DSNortsev** [comment]
https://github.com/ansible/ansible/pull/59219

**Pull request closed without merging**